internal/gps/source_manager.go

```diff
@@ -578,6 +578,18 @@ func (sm *SourceMgr) InferConstraint(s string, pi ProjectIdentifier) (Constraint
 		return version.Unpair(), nil
 	}
 
+	// Abbreviated git revision? If the string is present, there's a good shot of
+	// this.
+	if present, _ := sm.RevisionPresentIn(pi, Revision(s)); present {
```

Comment:
At the top of this function are a bunch of string checks to validate the revision. I was hoping that we could replace those heuristics with this new logic to lookup the revision from the repo.